python/ray/data/_internal/stats.py

```diff
@@ -411,13 +417,13 @@ def _stats_actor(self, create_if_not_exists=True) -> _StatsActor:
             self._stats_actor_handle is None
             or self._stats_actor_cluster_id != current_cluster_id
         ):
-            self._stats_actor_cluster_id = current_cluster_id
             if create_if_not_exists:
                 self._stats_actor_handle = _get_or_create_stats_actor()
             else:
                 self._stat_actor_handle = ray.get_actor(
                     name=STATS_ACTOR_NAME, namespace=STATS_ACTOR_NAMESPACE
                 )
+            self._stats_actor_cluster_id = current_cluster_id
```

Comment:
Move this after assigning actor handle because it's possible for `ray.get_actor` to fail